src/state/controls/Button.ts

```diff
@@ -71,4 +71,11 @@ export class Button extends Control<IButtonData> implements IButton {
     public giveInput(input: IButtonInput): Promise<void> {
         return this.sendInput(input);
     }
+
+    public update(changedData: Partial<IButtonData>): Promise<void> {
+        if (changedData.cooldown) {
```

Comment:
Connor, I want to enable batch updates.

I want to enable them easily. `Control.update(..what you want to update)` => Promise<void> (or <IControlData>) makes sense to me.

What doesn't is the cooldown. This is the only problem imo. As you initially were the one to update `setCooldown` to take the inputed duration and change it into a timestamp I would like you advisement on how to do this.  Within `control.update`.

I do not see how creating a utility method that allows people to update multiple properties in the same service call breaks Liskov.